frontend/src/components/PanelList.tsx

```diff
@@ -41,14 +41,15 @@ const marginForCheckbox = css`
 
 type Props = {
   schedule: Schedule;
+  datasourceID: number;
 };
 
 // TODO: Create PanelItem component and extract logic there.
-export const PanelList: FC<Props> = ({ schedule }) => {
+export const PanelList: FC<Props> = ({ schedule, datasourceID }) => {
```

Comment:
Just a style I've picked up, I like to do:

```
export type PanelList = React.FunctionalComponent<PanelListProps>;

export const PanelList: PanelList = ...
```

Benefit of this is that you can export the props used by a component, which is useful in typechecking variables in components higher up the tree. YMMV 😄 .

Reply:
aren't props checked anyway when passing them?

Reply:
Yeah, but I guess it's just for neatness in the calling code if maybe you want to "construct" the props in the higher up component, e.g. define an object for the props, somethinglike:

```
panelProps: PanelListProps = { ... };
return <PanelList {,,,panelProps} />
```

or whatever!